### ISSUE TITLE
Select JSON column Exception by using `query()` function

### ISSUE THREAD

**Issue by LibaoYangDev** [ClickHouse/clickhouse-connect#265]
### Describe the bug
When a select operation is performed using the `query()` function, an exception is thrown if the result contains complex JSON format.
### Steps to reproduce
1. create  a JSON table
```sql
CREATE TABLE IF NOT EXISTS test_table (id Int32, json JSON) ENGINE = MergeTree() ORDER BY id;
```
2. insert the complex data
```
:) select * from test_table Format JSONEachRow;

{"id":2,"json":{"content":"This is a description of the book","metadata":{"Author":"test","Number of pages":"","Title":"title","Year of Original Publication":"","Year of Publication":"2023"}}}
```
3. use `query()` function failed
```python
res = client.query("SELECT * FROM test_table")
```
```shell
clickhouse_connect.driver.exceptions.InternalError: Unrecognized ClickHouse type base: ofpages`String name: ofpages`String
```

**Comment by genzgd** [comment]
This bug is related to having spaces in your keys ('like a key of `Number of pages`).  ClickHouse parses that as a tuple element.  I'll see if I can improve the parser to handle that correctly, but the quickest workaround is to not use keys with spaces.

**Comment by genzgd** [comment]
This should be fixed in 0.6.19, although any time you have to deal with spaces in a column name there's always a chance of weird bugs.
